core/src/main/scala/io/finch/streaming/DecodeStream.scala

```diff
@@ -0,0 +1,48 @@
+package io.finch.streaming
+
+import java.nio.charset.Charset
+
+import com.twitter.io.Buf
+import io.finch.Application
+import scala.annotation.implicitNotFound
+
+/**
+  * Stream HTTP streamed payload represented as S[F, Buf] into
+  * a S[F, A] of arbitrary type `A`.
+  */
+trait DecodeStream[S[_[_], _], F[_], A] {
```

Comment:
What do you think about defining this under `io.finch` (no need for `streaming`)?

Reply:
Done